### PR TITLE
Support uploading large files and folders

### PR DESCRIPTION
**Pull request by oeway**
This PR adds chunked upload support for large files and folder uploads.

Specification: https://github.com/Studio-42/elFinder/wiki/Client-Server-API-2.1#upload

### CODE REVIEW

File: imjoy_elfinder/elfinder.py [hunk 1]
```diff
@@ -759,47 +767,102 @@ def __upload(self) -> None:
...
             max_size = self._options["upload_max_size"]
-            for name, data in up_files.items():
-                if name:
-                    name = self._check_utf8(name)
-                    total += 1
-                    name = os.path.basename(name)
+            chunk = self._request.get(API_CHUNK)
+            if chunk:
+                if chunk.endswith(".part"):
+                    start, end, total = [
```
Comment (oeway) [inline imjoy_elfinder/elfinder.py:773]: Verify the max upload size here

File: imjoy_elfinder/elfinder.py [hunk 2]
```diff
@@ -759,47 +767,135 @@ def __upload(self) -> None:
...
             max_size = self._options["upload_max_size"]
-            for name, data in up_files.items():
-                if name:
-                    name = self._check_utf8(name)
-                    total += 1
-                    name = os.path.basename(name)
-                    if not _check_name(name):
-                        self._set_error_data(name, "Invalid name: " + name)
+            chunk = self._request.get(API_CHUNK)
+            if chunk:
```
Comment (MartinHjelmare) [inline imjoy_elfinder/elfinder.py:771]: I would try to split this method up and see if that can make it easier to read.

Comment (oeway) [reply imjoy_elfinder/elfinder.py:771]: Yeah, the implementation and if statements are horrible at the moment, I try to make it work.

Still not perfect at the moment, when uploading large files, I cannot make the upload stop even if I set "warning". The current version silently shutdown if any error occurred, not sure how to solve yet.

We need some refactor with the upload function, could you please take a look see how to improve it? @MartinHjelmare

Comment (MartinHjelmare) [reply imjoy_elfinder/elfinder.py:771]: I can check later in the week.

Comment (oeway) [reply imjoy_elfinder/elfinder.py:781]: I will merge this for now, let's refactor it in another PR.

**Comment by oeway** [comment]
Having a lot of linter issue ;(

**Comment by MartinHjelmare** [comment]
Most problems seem to be that an argument is not allowed to be optional (None). Making sure the argument a string somehow will solve those.

**Comment by oeway** [comment]
> Most problems seem to be that an argument is not allowed to be optional (None) instead of a string. Making sure the argument a string somehow will solve those.

You are right, finally solved it by adding if checks.
Also splitted into two functions, but still there are space to improve, please take a look when you have time.

**Pull request merged**